MongoDB credentials should be url encoded

When setting up a mongodb connection, and you provide a password containing a colon `:`or at `@` symbol, the connection is not being created and the following stacktrace is thrown.

```
[22:41:20] ERROR: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at eu.thesimplecloud.launcher.external.module.handler.UnsafeModuleLoader.loadModuleClassInstance(UnsafeModuleLoader.kt:64)
        at eu.thesimplecloud.launcher.external.module.handler.UnsafeModuleLoader.loadModule(UnsafeModuleLoader.kt:53)
        at eu.thesimplecloud.launcher.application.ApplicationStarter.startApplication(ApplicationStarter.kt:58)
        at eu.thesimplecloud.launcher.startup.Launcher.startApplication(Launcher.kt:166)
        at eu.thesimplecloud.launcher.setups.StartSetup$onFinish$2.invoke(StartSetup.kt:49)
        at eu.thesimplecloud.launcher.setups.StartSetup$onFinish$2.invoke(StartSetup.kt:48)
        at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
Caused by: java.lang.IllegalArgumentException: The connection string contains invalid user information. If the username or password contains a colon (:) or an at-sign (@) then it must be urlencoded
        at com.mongodb.ConnectionString.<init>(ConnectionString.java:330)
        at eu.thesimplecloud.base.manager.database.MongoOfflineCloudPlayerHandler.getConnectionString(MongoOfflineCloudPlayerHandler.kt:82)
        at eu.thesimplecloud.base.manager.database.MongoOfflineCloudPlayerHandler.createMongoClient(MongoOfflineCloudPlayerHandler.kt:86)
        at eu.thesimplecloud.base.manager.database.MongoOfflineCloudPlayerHandler.<init>(MongoOfflineCloudPlayerHandler.kt:48)
        at eu.thesimplecloud.base.manager.startup.Manager.<init>(Manager.kt:170)
        ... 12 more
        ```

--- a/simplecloud-base/src/main/kotlin/eu/thesimplecloud/base/manager/database/MongoOfflineCloudPlayerHandler.kt
+++ b/simplecloud-base/src/main/kotlin/eu/thesimplecloud/base/manager/database/MongoOfflineCloudPlayerHandler.kt
@@ -40,6 +40,7 @@ import org.litote.kmongo.KMongo
 import org.litote.kmongo.find
 import org.litote.kmongo.findOne
 import org.litote.kmongo.getCollection
+import java.net.URLEncoder
 import java.util.*
 
 class MongoOfflineCloudPlayerHandler(val databaseConnectionInformation: DatabaseConnectionInformation) :
@@ -72,9 +73,9 @@ class MongoOfflineCloudPlayerHandler(val databaseConnectionInformation: Database
     private fun getConnectionString(): ConnectionString {
         val host = databaseConnectionInformation.host
         val port = databaseConnectionInformation.port
-        val databaseName = databaseConnectionInformation.databaseName
-        val userName = databaseConnectionInformation.userName
-        val password = databaseConnectionInformation.password
+        val databaseName = URLEncoder.encode(databaseConnectionInformation.databaseName,"utf-8")
+        val userName = URLEncoder.encode(databaseConnectionInformation.userName, "utf-8")
+        val password = URLEncoder.encode(databaseConnectionInformation.password,"utf-8")
         if (password.isBlank() || userName.isBlank()) {
             return ConnectionString("mongodb://$host:$port/?authSource=$databaseName")
         }

